### PR TITLE
Adjust tests for full Dart 2 runtime semantics; upgrade to SDK 2-dev.60 and enable --preview-dart-2 for example tests

### LINKED ISSUE #907: Upgrade to SDK 2-dev.60 and enable --preview-dart-2 for example tests

### PR DESCRIPTION
**Pull request by chalin**
Adjustments to example code and their tests:

- We can't run test code containing compile time errors anymore, so that induced some refactoring. Code excerpts with compile-time errors were moved to files that are only subject to analysis.
- With the full (or fuller) Dart 2 runtime semantics enabled, one of the tests changed because of [instantiate-to-bound](https://github.com/dart-lang/sdk/issues/33334#issuecomment-394755483) semantics.
- See other inlined comments below.

Staged, with the main changes occurring in this page:
- https://dartlang-org-staging-0.firebaseapp.com/guides/language/language-tour

cc @munificent 

Fixes #907

### CODE REVIEW

**Comment by chalin** [comment]
As usual, it will probably be easier to start the review by looking at the changes to the markdown files:
- src/_guides/language/language-tour.md
- src/_guides/language/effective-dart/usage.md

File: src/_guides/language/language-tour.md [hunk 1]
```diff
@@ -2149,11 +2149,10 @@ use the `rethrow` keyword.
 
 <?code-excerpt "misc/test/language_tour/exceptions_test.dart (rethrow)" replace="/rethrow;/[!$&!]/g"?>
 {% prettify dart %}
-final foo = '';
-
 void misbehave() {
   try {
-    foo = "You can't change a final variable's value.";
```
Comment (chalin) [inline src/_guides/language/language-tour.md]: Compile-time error prevents us from running this code. So I switch to code the would raise a runtime exception.

File: src/_guides/language/language-tour.md [hunk 2]
```diff
@@ -2217,13 +2216,11 @@ for a class. Constructor names can be either <code><em>ClassName</em></code> or
...
 <?code-excerpt "misc/test/language_tour/classes_test.dart (object-creation)" replace="/ as .*?;/;/g"?>
 {% prettify dart %}
-var jsonData = jsonDecode('{"x":1, "y":2}');
-
 // Create a Point using Point().
 var p1 = new Point(2, 2);
 
 // Create a Point using Point.fromJson().
-var p2 = new Point.fromJson(jsonData);
+var p2 = new Point.fromJson({'x': 1, 'y': 2});
```
Comment (chalin) [inline src/_guides/language/language-tour.md]: We had been using `jsonDecode('{"x":1, "y":2}');` but things were getting messy with the types. So I dropped the decoding (these examples are about object creation anyways), and just passed a "JSON literal" as an argument instead.

File: src/_guides/language/language-tour.md [hunk 3]
```diff
@@ -3342,35 +3340,34 @@ You can do this using `extends`.
...
 {% endprettify %}
 
-It's also OK to specify no generic argument:
+When you don't provide a type argument, it defaults to the type bound:
 
-<?code-excerpt "misc/lib/language_tour/generics/base_class.dart (no-generic-arg-ok)" replace="/Foo.\w+./[!$&!]/g"?>
+<?code-excerpt "misc/test/language_tour/generics_test.dart (no-generic-arg-ok)" replace="/expect\((.*?).toString\(\), .(.*?).\);/print($1); \/\/ $2/g"?>
 {% prettify dart %}
 var foo = new Foo();
+print(foo); // Instance of 'Foo<SomeBaseClass>'
```
Comment (chalin) [inline src/_guides/language/language-tour.md]: Here is where the instantiate-to-bounds semantics comes into play.

**Comment by chalin** [comment]
Travis CI blocked by https://github.com/dart-lang/pub/issues/1914.

File: src/_guides/language/language-tour.md [hunk 4]
```diff
@@ -3342,35 +3340,34 @@ You can do this using `extends`.
...
 
-<?code-excerpt "misc/lib/language_tour/generics/base_class.dart (SomeBaseClass-ok)" replace="/Foo.\w+./[!$&!]/g"?>
+<?code-excerpt "misc/test/language_tour/generics_test.dart (SomeBaseClass-ok)" replace="/Foo.\w+./[!$&!]/g"?>
 {% prettify dart %}
 var someBaseClassFoo = new [!Foo<SomeBaseClass>!]();
 var extenderFoo = new [!Foo<Extender>!]();
 {% endprettify %}
 
-It's also OK to specify no generic argument:
+When you don't provide a type argument, it defaults to the type bound:
```
Comment (kwalrath) [inline src/_guides/language/language-tour.md]: "type bound" threw me — we should either introduce that phrase at the top of this section or use words we've used before, such as "type you specified with `extends`"

Comment (kwalrath) [reply src/_guides/language/language-tour.md]: Actually, maybe we should use the original words:

It’s also OK to specify no generic argument:

Comment (chalin) [reply src/_guides/language/language-tour.md]: Ok, reverting.

**Pull request merged**